core/src/main/java/org/elasticsearch/search/suggest/completion/context/GeoContextMapping.java

```diff
@@ -254,7 +254,7 @@ protected GeoQueryContext fromXContent(QueryParseContext context) throws IOExcep
      */
     @Override
     public List<InternalQueryContext> toInternalQueryContexts(List<GeoQueryContext> queryContexts) {
-        List<InternalQueryContext> internalQueryContextList = new ArrayList<>();
+        List<InternalQueryContext> internalQueryContextList = new ArrayList<>(queryContexts.size());
```

Comment:
I think this one might be a step backwards though it is hard to tell without digging deeper.